ocs-api/import-owner-private-keys2.sh

```diff
@@ -0,0 +1,300 @@
+      chk $? 'generating ec certificate'
+      if [[ -f "$privateKey" ]]; then
+        openssl ec -in ecdsa"${var2}"private-key.pem -out ecdsa"${var2}"private-key.pem >/dev/null 2>&1
+        chk $? 'decrypting ec private key for macOS'
+      else
+        echo "No EC private key found"
+      fi
+    fi
+  fi
+  if [[ $keyType == "rsa" ]] || [ "$(uname)" != "Darwin" ]; then
+    (
+      echo "$COUNTRY_NAME"
+      echo "$STATE_NAME"
+      echo "$LOCALE_NAME"
+      echo "$COMPANY_NAME"
+      echo "$ORG_UNIT"
+      echo "$COMMON_NAME"
+      echo "$EMAIL_NAME"
+    ) | $gencmd >/dev/null 2>&1
+    chk $? 'generating rsa certificate'
+    if [ "$(uname)" == "Darwin" ] && [[ $keyType == "rsa" ]] && [[ -f "$privateKey" ]]; then
+        openssl rsa -in "$privateKey" -out "$privateKey" >/dev/null 2>&1
+        chk $? 'decrypting rsa private key for macOS'
+    fi
+  fi
+
+  if [[ -f $keyCert ]] && [[ -f "$privateKey" ]]; then
+    echo -e ""${keyType}" Private Key and "${keyType}"Key Certificate creation: Successful"
+    genPublicKey
+    popd >/dev/null
+  else
+    echo ""${keyType}" Private Key and "${keyType}"Key Certificate not found"
+    exit 2
+  fi
+}
+
+function genPublicKey() {
+  # This function is ran after the private key and owner certificate has been created. This function will create a public key to correspond with
+  # the owner private key/certificate. Generate a public key from the certificate file
+  openssl x509 -pubkey -noout -in $keyCert >"${keyType}"pub-key.pem
+  chk $? 'Creating public key...'
+  echo "Generating "${keyType}" public key..."
+  if [[ -f "${keyType}"pub-key.pem ]]; then
+    echo -e ""${keyType}" public key creation: Successful"
+    mv "${keyType}"pub-key.pem ..
+  else
+    echo -e ""${keyType}" public key creation: Unsuccessful"
+    exit 2
+  fi
+}
+
+function combineKeys() {
+  #This function will concatenate all public keys into one
+  if [[ -f "rsapub-key.pem" && -f "ecdsa256pub-key.pem" && -f "ecdsa384pub-key.pem" ]]; then
+    #Combine all the public keys into one
+    if [[ -f owner-public-key.pem ]]; then
+      rm owner-public-key.pem
+    fi
+    #adding delimiters for SDO 1.9 pub key format
+    echo "," >> rsapub-key.pem && echo "," >> ecdsa256pub-key.pem
+    echo "Concatenating Public Key files..."
+    for i in "rsapub-key.pem" "ecdsa256pub-key.pem" "ecdsa384pub-key.pem"
+      do
+        local keyName=$i
+        local removeWord="pub-key.pem"
+        keyName=${keyName//$removeWord/}
+        #adding delimiters for SDO 1.9 pub key format
+        echo ""$keyName":" >key.txt
+        #make sure key name is lowercase for file name
+        cat key.txt $i >> ${ORG_UNIT}_${KEY_NAME}_public-key.pem
+      done
+    chk $? 'Concatenating Public Key files...'
+    rm -- ecdsa*.pem && rm rsapub* && rm key.txt
```

Comment:
Tip: you can give multiple files to a single `rm` cmd, no need to and multiple `rm` cmds together